File: julia/FizzBuzz/test/FizzBuzz.jl
#
# @brief FizzBuzz kata unit tests.
#

@test FizzBuzz.answer_for(1) == "1"
@test FizzBuzz.answer_for(2) == "2"
@test FizzBuzz.answer_for(3) == "Fizz"
@test FizzBuzz.answer_for(4) == "4"
@test FizzBuzz.answer_for(5) == "Buzz"
@test FizzBuzz.answer_for(9) == "Fizz"
@test FizzBuzz.answer_for(10) == "Buzz"
@test FizzBuzz.answer_for(15) == "FizzBuzz"
@test FizzBuzz.answer_for(90) == "FizzBuzz"

@test FizzBuzz.answer_for(23) == "Fizz"
@test FizzBuzz.answer_for(52) == "Buzz"
@test FizzBuzz.answer_for(53) == "FizzBuzz"

File: javascript/fizzbuzz/fizzbuzz.js
/**
 * @brief Indicates wheter the number is in Fizz set.
 *
 * @param  {Integer} number The number to test.
 * @return {Boolean}        True if the number is in the set; otherwise False.
 */
function isFizz (number) {
  return (number % 3) === 0;
}

/**
 * @brief Indicates wheter the number is in Buzz set.
 *
 * @param  {Integer} number The number to test.
 * @return {Boolean}        True if the number is in the set; otherwise False.
 */
function isBuzz (number) {
  return (number % 5) === 0;
}

/**
 * @brief Indicates wheter the String is empty.
 *
 * @param  {String}  str The string to test.
 * @return {Boolean}     True if the string is empty; otherwise False.
 */
function isEmpty (str) {
  return str === '';
}

/**
 * @brief Computes the FizzBuzz game result for the given number.
 *
 * @param  {Integer} number The number to play.
 * @return {Boolean}        The FizzBuzz game result.
 */
module.exports.compute = function (number) {
  var result = '';

  if (isFizz(number))
    result += 'Fizz';

  if (isBuzz(number))
    result += 'Buzz';

  return isEmpty(result) ? number.toString() : result;
}

